termbox/compat.go

```diff
@@ -109,6 +124,10 @@ func fixColor(c tcell.Color) tcell.Color {
 	return c
 }
 
+func MakeStyle(fg, bg Attribute) tcell.Style {
```

Comment:
So I'm not sure why you'd need this -- your using a tcell.Style, but termbox knows nothing about this.  This really blurs the boundary between the two APIs.  Probably this is so that you can start shifting more gradually from termbox to tcell, but I'm a bit concerned about potential namespace issues later.  For example, what if termbox later defines an API by the same name.  This could collide, and we'd have no way to offer compatibility.

Thinking about this, I think another package namespace for this sort of thing is appropriate.  Or, perhaps we can stick it in tcell directly, "MakeTermboxStyle()", although that means teaching tcell about termbox.Attribute.

Reply:
Hi Garrett!

I didn't want to make any of these changes to compat.go.  I'd would prefer not to have to, in fact!

I only wanted to get a project that was using termbox-go ported (over to tcell). 

So the easiest way to come to a meeting of minds would be to have you look at the `gemacs` project code (https://github.com/glycerine/gemacs) and tell me if there's another way -- just because I couldn't find one, maybe you can!  For an existing project that uses termbox-go, compat.go is a great (perhaps only sane) way to get started with tcell.

`gemacs` is a nice, small, self-contained project, pure Go so easy to build.

Here are where MakeStyle is currently used in `gemacs`:

~~~
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/gemacs (master) $ ack MakeStyle
vendor/github.com/gdamore/tcell/termbox/compat.go
127:func MakeStyle(fg, bg Attribute) tcell.Style {

vendor/github.com/nsf/tulib/buffer.go
76:             st := termbox.MakeStyle(proto.Fg, proto.Bg)
152:                            termbox.MakeStyle(from.Fg, from.Bg))
178:    st := termbox.MakeStyle(proto.Fg, proto.Bg)
195:    st := termbox.MakeStyle(params.Fg, params.Bg)
221:    st := termbox.MakeStyle(params.Fg, params.Bg)
277:    st := termbox.MakeStyle(params.Fg, params.Bg)

view.go
284:                            st := termbox.MakeStyle(termbox.ColorDefault, termbox.ColorDefault)
305:                                            st := termbox.MakeStyle(cell.Fg, cell.Bg)
312:                    red := termbox.MakeStyle(termbox.ColorRed, termbox.ColorDefault)
345:                                    st := termbox.MakeStyle(cell.Fg, cell.Bg)
362:                    st := termbox.MakeStyle(termbox.ColorDefault, termbox.ColorDefault)
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/gemacs (master) $
~~~

e.g. https://github.com/glycerine/gemacs/blob/master/view.go#L284

Here is where PollEventOnScreen is used:
~~~
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/gemacs (master) $  ack PollEventOnScreen
gemacs.go
433:                    g.termbox_event <- termbox.PollEventOnScreen(g.uibuf.Screen)

vendor/github.com/gdamore/tcell/termbox/compat.go
368:func PollEventOnScreen(s tcell.Screen) Event {
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/gemacs (master) $
~~~
e.g. https://github.com/glycerine/gemacs/blob/master/gemacs.go#L433